### PR TITLE
updated technical documentation: domain model, classification + tagging, statemachines

### PR DESCRIPTION
**Pull request by techsmyth**
Updated domain model for innovation flow addition, improved tagsets.
Adds technical note for classification + tagging
Adds technical note for state machines usage

### CODE REVIEW

File: docs/classification-tagging.md [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+The TagsetTemplate entity is held in a containing TagsetTemplateSet entity. This is both to make it easier to use in multiple places in the domain model, and also avoids ORM linkages in the code base (technical).
+
+The Collaboration entity uses a TagsetTemplateSet to hold the TagsetTemplates to be applied to all Callouts in that Collaboration. The following TagsetTemplates are used per Journey type:
+* Space: 
+    * Default: type = freeform, for allowing any tags to be added
+    * DisplayLocation: type = singleSelect, for specifying the palcement of the Callout within the display of the Space
+* Challenge: 
+    * Default: type = freeform, for allowing any tags to be added    
+    * DisplayLocation: type = singleSelect, for specifying the palcement of the Callout within the display of the Space
```
Comment (valentinyanakiev) [inline docs/classification-tagging.md]:
```suggestion
    * DisplayLocation: type = singleSelect, for specifying the placement of the Callout within the display of the Space
```

File: docs/classification-tagging.md [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    * States: type = singleSelect for the set of visible states from the InnovationFlow
+
+Then each time a Callout is added to the Collaboration, it's profile will have the above Tagsets created as well as the Default tagset.
```
Comment (valentinyanakiev) [inline docs/classification-tagging.md]:
```suggestion
Then each time a Callout is added to the Collaboration, its profile will have the above Tagsets created as well as the Default tagset.  
```

File: docs/classification-tagging.md [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+Tagsets added to an entity to be specified by the parent entity via passing in the applicable list of TagsetTemplates. 
+For example for Callouts that should be the Collaboration
+This does mean that the Collaboration entity needs to know what "Tagset Templates" to apply to each newly created Callout, which it then manages via its TagsetTemplateSet.
+
+**Updating of TagsetTemplates**
+
+It is expected that the set of values in a single / multiple select Tagset do not change frequently. 
+
+For example the set of displayLocation values is fairly fixed for the Space and Challenge. 
+When the set of allowedValues does need to change, for example if the InnovationFlow lifecycle being used changes, then the business logic will need to ensure that relevant TagsetTemplate is updated - plus ensuring that the selected value is one of th evalues in the set of allowedValues.
```
Comment (valentinyanakiev) [inline docs/classification-tagging.md]:
```suggestion
When the set of allowedValues does need to change, for example, if the InnovationFlow lifecycle being used changes, then the business logic will need to ensure that the relevant TagsetTemplate is updated - plus ensuring that the selected value is one of the values in the set of allowedValues. 
```

**Review by valentinyanakiev — CHANGES_REQUESTED** [review]

**Comment by techsmyth** [comment]
@valentinyanakiev thanks for the text fixes, they are applied so please go ahead + merge

**Review by valentinyanakiev — APPROVED** [review]

**Pull request merged**